otelcol/collector.go

```diff
@@ -202,6 +207,12 @@ func (col *Collector) setupConfigurationComponents(ctx context.Context) error {
 		return err
 	}
 
+	if col.ol != nil {
+		for _, log := range col.ol.All() {
+			col.service.Logger().Log(log.Level, log.Message)
```

Comment:
I am wondering if we should use the core directly here via its `Check` and `Write` methods. That way we would preserve the exact logged message, including the timestamp and stack trace. I think it can be done, but it's not trivial since there is some logic we would have to copy from [here](https://github.com/uber-go/zap/blob/ed52ec03fc86806389deaf551500e7a48214ca62/logger.go#L347-L356).

Do you think it's worth the effort?

Reply:
>  I dont think stacktraces is a valid scenario bc currently if the confmap errors its logs will not get written.

I don't get what you mean by this. What I meant is that instead of having `otelcol@v0.98.0/collector:XX` we would have the actual line where this log came from.

Reply:
Looking at #10056 this is actually necessary since we are not logging the fields (if you compare the screenshots, with this PR you don't know what env var is unset)

Reply:
I really don't like the idea of duplicating functionality that exists in `zapcore` already

Reply:
Also I can fix the field issue by passing in `log.Context...` as the last param.

Reply:
> I really don't like the idea of duplicating functionality that exists in `zapcore` already

To be clear: it does not exist in zapcore, it exists in `*zap.Logger`